R/DataDiscoveryRetrieval.R

```diff
@@ -324,146 +351,176 @@ TADA_DataRetrieval <- function(startDate = "null",
           # Return sf
           arcgislayers::arc_select() %>%
           # If a value provided, then filter
-          {if ((tribe_name_parcel != "null") & (tribe_name_parcel != "null")) {
-            filter(., TRIBE_NAME %in% tribe_name_parcel)
-          } else {
-            .
-          }}
-        
+          {
+            if (all(tribe_name_parcel != "null")) {
+              dplyr::filter(., TRIBE_NAME %in% tribe_name_parcel)
+            } else {
+              .
+            }
+          }
+
         # Otherwise filter by PARCEL_NO (Note that values in this col are not unique)
-      } else if(tribal_area_type == "Alaska Native Allotments"){
-        
-        aoi_sf <- filter(map_service_urls,
-                         tribal_area == tribal_area_type)$url %>%
+      } else if (tribal_area_type == "Alaska Native Allotments") {
+        aoi_sf <- dplyr::filter(
+          map_service_urls,
+          tribal_area == tribal_area_type
+        )$url %>%
           arcgislayers::arc_open() %>%
           arcgislayers::arc_select() %>%
-          {if ((tribe_name_parcel != "null") & (tribe_name_parcel != "null")) {
-            filter(., PARCEL_NO %in% tribe_name_parcel)
-          } else {
-            .
-          }}
-        
+          {
+            if (all(tribe_name_parcel != "null")) {
+              dplyr::filter(., PARCEL_NO %in% tribe_name_parcel)
+            } else {
+              .
+            }
+          }
       } else {
-        stop("Tribal area type not recognized. Refer to TADA_TribalOptions() for query options.")
+        stop("Tribal area type or tribal name parcel not recognized. Refer to TADA_TribalOptions() for query options.")
       }
-      
     }
-    
+
     # Check and/or fix geometry
     aoi_sf <- sf::st_make_valid(aoi_sf)
-    
+
     # Match CRS
-    if(sf::st_crs(aoi_sf) != 4326){
+    if (sf::st_crs(aoi_sf) != 4326) {
       aoi_sf <- sf::st_transform(aoi_sf, crs = 4326)
     }
-    
+
     # Get bbox of the sf object
     input_bbox <- sf::st_bbox(aoi_sf)
-    
+
     # Query info on available data within the bbox
-    bbox_avail <- suppressMessages(
-      dataRetrieval::whatWQPdata(
-        WQPquery,
-        bBox = c(input_bbox$xmin, input_bbox$ymin, input_bbox$xmax, input_bbox$ymax)
-      )
+
+    # Try getting WQP info
+    message("Checking what data is available. This may take a moment.")
+
+    # Don't want to print every message that's returned by WQP
+    quiet_bbox_avail <- quiet_whatWQPdata(
+      WQPquery,
+      bBox = c(input_bbox$xmin, input_bbox$ymin, input_bbox$xmax, input_bbox$ymax)
     )
-    
+
+    # Alert & stop if an http error was received
+    if (is.null(quiet_bbox_avail$result)) {
+      stop_message <- quiet_bbox_avail$messages %>%
+        grep(pattern = "failed|HTTP", x = ., ignore.case = FALSE, value = TRUE) %>%
+        paste("\n", ., collapse = "") %>%
+        paste("The WQP request returned a NULL with the following message(s): \n",
+          .,
```

Comment:
```suggestion
          .,
          "The bounding box is likely too large for this process. Reduce your area of interest and try again.",
```

Okay, this one is still tripping up, so I suggest a different message here. Here's the code that would be hard to figure out why it would fail:

```
areas <- TADA_TribalOptions(tribal_area_type = "American Indian Reservations")
test <- TADA_DataRetrieval(
  startDate = "2015-01-01",
  tribal_area_type = "American Indian Reservations", 
  tribe_name_parcel = areas$TRIBE_NAME[1:3],
  ask = TRUE)
```

Turns out the bbox is 30 deg by 12 deg. oops.